pytest 9.1 changes

Our "basic" pip test has just started to throw warning/errors like

```
>       assert 0 == nwarnings
E       assert 0 == 1

work/sherpa/sherpa/sherpa/conftest.py:305: AssertionError
--------------------------- Captured stdout teardown ---------------------------
*** Warnings created: 1
1/1 {message : PytestDeprecationWarning("\nModule 'sherpa.astro.xspec._xspec' was found, but when imported by pytest it raised:\n    ImportError('XSPEC support is not enabled')\nIn pytest 9.1 this warning will become an error by default.\nYou can fix the underlying problem, or alternatively overwrite this behavior and silence this warning by passing exc_type=ImportError explicitly.\nSee https://docs.pytest.org/en/stable/deprecations.html#pytest-importorskip-default-behavior-regarding-importerror"), category : 'PytestDeprecationWarning', filename : '<doctest sherpa.astro.io.xstable[0]>', lineno : 1, line : None}
_______ ERROR at teardown of [doctest] sherpa.astro.io.xstable.BaseParam _______
```

Need to sit down with a good book and stare at https://docs.pytest.org/en/stable/deprecations.html#pytest-importorskip-default-behavior-regarding-importerror until it gives up it's secrets, since the fact this comes from a `doctest` means t's not at all obvious where to stick the `exc_type=ImportError` argument...

--- a/pytest.ini
+++ b/pytest.ini
@@ -47,6 +47,7 @@ doctest_norecursedirs =
     sherpa/astro/tests
     sherpa/astro/utils
     sherpa/astro/xspec
+    sherpa/astro/io/xstable.py  # see discussion in https://github.com/sherpa/sherpa/issues/2421
     sherpa/conftest.py
     sherpa/estmethods
     sherpa/image

